cmd/imagec/docker.go

```diff
@@ -269,9 +257,35 @@ func FetchImageBlob(options ImageCOptions, image *ImageWithMeta) (string, error)
 
 	diffID = fmt.Sprintf("sha256:%x", diffIDSum.Sum(nil))
 
+	// this isn't an empty layer, so we need to calculate the size
 	if diffID != string(DigestSHA256EmptyTar) {
-		// bytesRead represents the size of the decompressed layer data
-		image.size = bytesRead
+		var layerSize int64
+
+		// seek to the beginning of the file
+		imageFile.Seek(0, 0)
+
+		// recreate the decompressed tar Reader
+		decompressedTar, err := archive.DecompressStream(imageFile)
+		if err != nil {
+			return "", err
+		}
+
+		// get a tar reader for access to the files in the archive
+		tr := tar.NewReader(decompressedTar)
+
+		// iterate through tar headers to get file sizes
+		for {
+			tarHeader, err := tr.Next()
+			if err == io.EOF {
+				break
+			}
+			if err != nil {
+				return "", err
+			}
+			layerSize += tarHeader.Size
+		}
+
+		image.size = layerSize
 	}
```

Comment:
`DecompressStream` wraps the gzipped `File` in a `gzip.Reader`, so (as I understand Readers & Writers) the bytes are streaming from the compressed file Reader, through `gzip.Reader` where they are decompressed and into the tar Reader, where bytes are streamed into `/dev/null` until the next file header in the tar is hit.

Does that sound crazy? I'm like you.. trying to understand and not confident to assert with 100% certainty that what I suspect is happening is actaully happening. :)